server/src/main/java/org/elasticsearch/cluster/coordination/MasterHistoryService.java

```diff
@@ -132,26 +164,27 @@ public void onFailure(Exception e) {
                 @Override
                 public void onFailure(Exception e) {
                     logger.warn("Exception connecting to master node", e);
-                    remoteHistoryOrException = new RemoteHistoryOrException(e);
+                    remoteHistoryOrException = new RemoteHistoryOrException(e, currentTimeMillisSupplier.getAsLong());
                 }
             }
         );
     }
 
-    record RemoteHistoryOrException(List<DiscoveryNode> remoteHistory, Exception exception) { // non-private for testing
+    // non-private for testing
+    record RemoteHistoryOrException(List<DiscoveryNode> remoteHistory, Exception exception, long creationTimestamp) {
 
         public RemoteHistoryOrException {
             if (remoteHistory != null && exception != null) {
                 throw new IllegalArgumentException("Remote history and exception cannot both be non-null");
             }
         }
 
-        RemoteHistoryOrException(List<DiscoveryNode> remoteHistory) {
-            this(remoteHistory, null);
+        RemoteHistoryOrException(List<DiscoveryNode> remoteHistory, long creationTimestamp) {
```

Comment:
Can we rename the creationTimestamp to include the unit everywhere? ( https://github.com/elastic/elasticsearch/pull/86936/files#r880273628 )